From dfc692350a04a70b4ca65667c30869b3bfdaf034 Mon Sep 17 00:00:00 2001
Message-Id: <dfc692350a04a70b4ca65667c30869b3bfdaf034.1374788346.git.crobinso@redhat.com>
From: Peter Krempa <pkrempa@redhat.com>
Date: Tue, 16 Jul 2013 15:39:06 +0200
Subject: [PATCH] qemu: Fix double free of returned JSON array in
 qemuAgentGetVCPUs()

A part of the returned monitor response was freed twice and caused
crashes of the daemon when using guest agent cpu count retrieval.

 # virsh vcpucount dom --guest

Introduced in v1.0.6-48-gc6afcb0
---
 src/qemu/qemu_agent.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/qemu/qemu_agent.c b/src/qemu/qemu_agent.c
index aca5ff3..72bf211 100644
--- a/src/qemu/qemu_agent.c
+++ b/src/qemu/qemu_agent.c
@@ -1529,7 +1529,6 @@ qemuAgentGetVCPUs(qemuAgentPtr mon,
 cleanup:
     virJSONValueFree(cmd);
     virJSONValueFree(reply);
-    virJSONValueFree(data);
     return ret;
 }
 
-- 
1.8.3.1

